lib/http/cookie_jar/yaml_saver.rb

```diff
@@ -73,13 +72,9 @@ def default_options
     {}
   end
 
-  if YAML.name == 'Psych' && Psych::VERSION >= '3.1'
-    def load_yaml(yaml)
-      YAML.safe_load(yaml, :permitted_classes => %w[Time HTTP::Cookie Mechanize::Cookie DomainName], :aliases => true)
-    end
-  else
-    def load_yaml(yaml)
-      YAML.load(yaml)
-    end
+  def load_yaml(yaml)
```

Comment:
I rewrote this method to avoid negating the [requested introduction of `autoload :YAML`](https://github.com/sparklemotion/http-cookie/pull/43#pullrequestreview-1505731171). It adds an extra level of indirection to every load, but only on extremely old and unsupported rubies.